### DISCUSSION TITLE
Widget disappears when removed and mounted again

### DISCUSSION THREAD

**Discussion by davidbrochart** [Textualize/textual#1445] [Q&A] [answered]
In the following example, a widget mounts a button when `s` is pressed the first time. When `s` is pressed again, the widget is removed and mounted again, but the button disappears.
Any idea why?

```py
from textual.app import App, ComposeResult
from textual.widget import Widget
from textual.widgets import Button, Footer


class MyWidget(Widget):

    def __init__(self):
        self.button = Button()
        super().__init__()

    def compose(self) -> ComposeResult:
        yield self.button


class MyApp(App):

    BINDINGS = [("s", "show_button", "Show Button")]

    def __init__(self):
        self.my_widget = None
        super().__init__()

    def compose(self) -> ComposeResult:
        yield Footer()

    async def action_show_button(self) -> None:
        if self.my_widget is None:
            self.my_widget = MyWidget()
        else:
            await self.my_widget.remove()
        await self.mount(self.my_widget)


if __name__ == "__main__":
    app = MyApp()
    app.run()
```

**Comment by davep** [comment, marked as answer]
To the best of my knowledge, when you `remove` a widget you're, in effect, destroying it (for want of a better term). Once you `remove` a particular instance you can't mont it again. If you want to add a button again you'll want to `mount` a fresh instance of `MyWidget`.

**Reply by willmcgugan** [reply]
Yeah, Widgets that have been removed are in a shutdown state. I never intended for widgets to be mounted again after adding.

Not crazy to try of course. So I think we should do one of two things: A) Make it work or B) Raise an "Can't mount shutdown widget" error or something.

@davidbrochart You probably know this, but you can set `display=True` to hide a widget without removing it.

**Reply by davep** [reply]
To correct what seems to be a typo there, [`display="none"`](https://textual.textualize.io/styles/display/).

**Reply by willmcgugan** [reply]
Also available as a property, so `my_widget.display=False` works.

**Reply by davep** [reply]
Surely you don't expect me to scroll *all* the way to the bottom of the docs I link to while I'm on holiday? ¯\\_(ツ)_/¯

**Reply by davidbrochart** [reply]
Thanks, I think using the `display` property is better indeed. BTW, is there a means to know if a widget is mounted?

**Reply by willmcgugan** [reply]
`App.is_mounted(my_widget)`

**Reply by davidbrochart** [reply]
Thanks @willmcgugan and @davep !
